### PR TITLE
Update to obsproc.v1.0.2 and prepobs.v1.0.1

### LINKED ISSUE #1033: Update to obsproc.v1.0.2 and prepobs.v1.0.1 in develop
**Description**

The `develop` branch is using an outdated version of the obsproc and prepobs packages. The obsproc and prepobs systems had major updates in the port to WCOSS2, including turning prepobs into a module that is loaded at runtime.

Refs #419 

Make the following updates in `develop`:

- [x] 1. Replace `HOMEobsproc_prep`, `HOMEobsproc_network` with `HOMEobsproc`; remove `HOMEobsproc_global` variable.
- [x] 2. Set `HOMEobsproc` to the local installs of `obsproc.v1.0.2` (depending on timing, may move to the newer `obsproc.v1.1.0` which is being handed off with GFSv16.3).
- [x] 3. Add module loads for `prepobs.v1.0.1` into `module_base` modulefiles.
- [x] 4. Replace `$HOMEobsproc_network/jobs/JGLOBAL_PREP` in `jobs/rocoto/prep.sh` with `$HOMEobsproc/jobs/JOBSPROC_GLOBAL_PREP`.
- [x] 5. Additionally...retire `config.prepbufr`. Upon examination, the `config.prepbufr` and its only use to set `GESROOT` in the prep job is no longer needed. Delete `config.prepbufr` and remove it from the config list in `jobs/rocoto/prep.sh`.

When version files and version variables are introduced into `develop` the `HOMEobsproc` variable will be set using `obsproc_ver` in version files and `prepobs_ver` used in the modulefiles will also be set in version files.

**Acceptance Criteria (Definition of Done)**

Prep jobs runs successfully with new obsproc/prepobs versions and produces expected files (e.g. prepbufr). Check if resulting prepbufr files differ from prepbufr produced from current obsproc package in `develop`.

**Dependencies**

- [x] Local installs of new obsproc and prepobs versions are already in place on supported platforms.

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
**Description**

This PR updates the `develop` branch to the latest obsproc and prepobs packages that resulted from the WCOSS2 port (GFSv16): `obsproc.v1.0.2` and `prepobs.v1.0.1`. This PR also includes some prep job cleanup and fixes.

Resolves #1033

Changes:

1. Add `launcher_PREP` variable to `HERA.env` and `ORION.env`; equals "srun" on those platforms, already set in `WCOSS2.env`.
2. Update `jobs/rocoto/prep.sh` to invoke new obsproc JJOB: replace `$HOMEobsproc_network/jobs/JGLOBAL_PREP` with `$HOMEobsproc/jobs/JOBSPROC_GLOBAL_PREP`
3. Add prepobs module load to `module_base.hera.lua` and `module_base.orion.lua` (already in `module_base.wcoss2.lua`); the prepobs package is now accessed with a module load.
4. Update `HOMEobsproc` variables in `config.base.emc.dyn` and `config.base.nco.static`: replace `HOMEobsproc_prep`, `HOMEobsproc_network` and `HOMEobsproc_global` with `HOMEobsproc`.
5. Update `config.prep` to add needed `cdate10`, `COMPONENT`, and `COMINtcvital` variables.
6. Retire prepbufr config (`config.prepbufr`) and reference in `config.resources` prep section.

**Type of change**

- [x New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] Cycled tests on Hera, Orion, WCOSS2

Testing on Hera and Orion reproduced control tests (identical prepbufr files).
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: parm/config/config.prep [hunk 1]
```diff
@@ -17,6 +18,8 @@ export DO_RELOCATE="NO"
 export TROPCYQCRELOSH="$HOMEgfs/scripts/exglobal_atmos_tropcy_qc_reloc.sh"
 export SENDCOM=YES
 
+export COMPONENT=${COMPONENT:-atmos}
```
Comment (aerorahul) [inline parm/config/config.prep]: Is this really something that will be set from above?
Isn't component here "always" `atmos`?

File: parm/config/config.prep [hunk 2]
```diff
@@ -9,6 +9,7 @@ echo "BEGIN: config.prep"
 . $EXPDIR/config.resources prep
 
 export DO_MAKEPREPBUFR="YES"   # if NO, will copy prepbufr from globaldump
+export cdate10=${PDY}${cyc}
```
Comment (aerorahul) [inline parm/config/config.prep:12]: Who is needing this variable to be set in their env.?
Is production setting this in `ecf` of the obsprocs?
Just curious.

**Review by aerorahul — APPROVED** [review]
looks good.
just a couple of minor comments.

File: parm/config/config.prep (hunk 2 above)
Comment (KateFriedman-NOAA) [reply parm/config/config.prep:12]: We have to set `cdate10` for obsproc `scripts/exglobal_makeprepbufr.sh`:

```
cdate10=${cdate10:-`cut -c7-16 ncepdate`}
```

https://github.com/NOAA-EMC/obsproc/blob/develop/scripts/exglobal_makeprepbufr.sh#L78

I had them add the override this year so we can pass `cdate10` in for development runs. In ops the default (`ncepdate`) is used. The same update was made in the GFSv16 system.

**Review by aerorahul — APPROVED** [review]
looks good.
thanks~

**Pull request merged**